crates/external_services/src/file_storage/aws_s3.rs

```diff
@@ -0,0 +1,157 @@
+use aws_config::meta::region::RegionProviderChain;
+use aws_sdk_s3::{
+    operation::{
+        delete_object::DeleteObjectError, get_object::GetObjectError, put_object::PutObjectError,
+    },
+    Client,
+};
+use aws_sdk_sts::config::Region;
+use common_utils::{errors::CustomResult, ext_traits::ConfigExt};
+use error_stack::{IntoReport, ResultExt};
+use futures::TryStreamExt;
+use storage_impl::errors::ApplicationError;
+
+static AWS_FILE_STORAGE_CLIENT: tokio::sync::OnceCell<AwsFileStorageClient> =
+    tokio::sync::OnceCell::const_new();
+
+/// Configuration for AWS S3 file storage.
+#[derive(Debug, serde::Deserialize, Clone, Default)]
+#[serde(default)]
+pub struct AwsFileStorageConfig {
+    /// The AWS region to send file uploads
+    pub region: String,
+    /// The AWS s3 bucket to send file uploads
+    pub bucket_name: String,
+}
+
+impl AwsFileStorageConfig {
+    /// Retrieves the AWS S3 file storage client, initializing it if necessary.
+    #[inline]
+    pub async fn get_aws_file_storage_client(&self) -> &'static AwsFileStorageClient {
+        AWS_FILE_STORAGE_CLIENT
+            .get_or_init(|| AwsFileStorageClient::new(self))
+            .await
+    }
```

Comment:
I'd prefer returning the client itself instead of returning a `OnceCell`. We can store the client behind an `Arc` in the `AppState`.

Is there any specific reason for using `OnceCell` otherwise?

Reply:
It's better to use AppState, currently this is being used for Disputes alone. But, in future we are going to reuse for other use cases as well.